src/Symfony/Component/PropertyAccess/Mapping/ClassMetadata.php

```diff
@@ -0,0 +1,126 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\PropertyAccess\Mapping;
+
+/**
+ * {@inheritdoc}
+ *
+ * @author Luis Ramón López <lrlopez@gmail.com>
+ */
+class ClassMetadata
+{
+    /**
+     * @var string
+     *
+     * @internal This property is public in order to reduce the size of the
+     *           class' serialized representation. Do not access it. Use
+     *           {@link getName()} instead.
+     */
+    public $name;
+
+    /**
+     * @var PropertyMetadata[]
+     *
+     * @internal This property is public in order to reduce the size of the
+     *           class' serialized representation. Do not access it. Use
+     *           {@link getPropertiesMetadata()} instead.
+     */
+    public $propertiesMetadata = array();
```

Comment:
I think the name of this property and its getter is a bit weird. We have a collection of "property metadata". The plural of that shouldn't be "properties metadata". Unfortunately, the plural of "metadata" is still "metadata".

Ideas?

We could solve this like in the Validator component by adding two methods: `getPropertyMetadata($property)` and `getProperties()` (returning property names).